File: go/space-age/space_age.go
package space

// Planet is a string with the name of a Planet.
type Planet string

// orbitalPeriods is map of the planets and their orbital periods.
var orbitalPeriods map[Planet]float64 = map[Planet]float64{
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Earth":   1.0,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
}

// secondsInEarthYear is the number of seconds in an Earth year that's 365.25 years long.
var secondsInEarthYear float64 = 365.25 * 24 * 60 * 60

// Age returns a person's age on a given planet.
func Age(seconds float64, planet Planet) float64 {
	orbitalPeriod, found := orbitalPeriods[planet]

	if !found {
		return -1.0
	}

	age := seconds / (orbitalPeriod * secondsInEarthYear)

	return age
}

File: c/space-age/space_age.c
#include "space_age.h"

const float seconds_in_earth_year = 365.25F * 24 * 60 * 60;

bool is_planet(planet_t planet) {
    if (planet >= MERCURY && planet < PLANET_COUNT) {
        return true;
    }

    return false;
}

float age(planet_t planet, int64_t seconds) {
    if (!is_planet(planet)) {
        return -1.0F;
    }

    float planets[PLANET_COUNT] = {0.0F};

    planets[MERCURY] = 0.2408467F;
    planets[VENUS] = 0.61519726F;
    planets[EARTH] = 1.0F;
    planets[MARS] = 1.8808158F;
    planets[JUPITER] = 11.862615F;
    planets[SATURN] = 29.447498F;
    planets[URANUS] = 84.016846F;
    planets[NEPTUNE] = 164.79132F;

    return (float)seconds / (planets[planet] * seconds_in_earth_year);
}
